numcodecs/zarr3.py

```diff
@@ -56,6 +58,23 @@
 CODEC_PREFIX = "numcodecs."
 
 
+def _from_zarr_dtype(dtype: Any) -> np.dtype:
+    """
+    Get a numpy data type from an array spec, depending on the zarr version.
+    """
+    if Version(version('zarr')) >= Version("3.1.0"):
+        return dtype.to_native_dtype()
+    return dtype  # pragma: no cover
+
+
+def _to_zarr_dtype(dtype: np.dtype) -> Any:
```

Comment:
Same here - shouldn't it only return subclasses of zarr data types?

Reply:
https://github.com/zarr-developers/zarr-python/blob/a26926cf4ec51a12ca28ff31c13ef2c8a32d51f9/src/zarr/core/dtype/__init__.py#L191